colossus/src/main/scala/colossus/protocols/http/Header.scala

```diff
@@ -113,25 +122,40 @@ object TransferEncoding {
     val value = "chunked"
   }
 
-  case object Gzip extends TransferEncoding {
+  private val all = Seq(Identity, Chunked)
+  def unapply(str : String) : Option[TransferEncoding] = {
+    all.find(_.value == str.toLowerCase)
+  }
+}
+
+sealed trait ContentEncoding {
+  def value: String
+}
+object ContentEncoding {
```

Comment:
Yeah, I started to actually add more functionality for this, but backed off since it was less trivial than I first thought.